app/src/backend/environment/utils.ts

```diff
@@ -73,16 +73,16 @@ const processObject = (input: Record<string, any>, variables: Variables): Render
   );
 
   return {
-    renderedTemplate: result.rendered,
+    renderedTemplate: result.rendered as T,
     usedVariables: result.variables,
   };
 };
 
-const processTemplateString = (input: string, variables: Variables): RenderResult => {
+const processTemplateString = <T extends string>(input: T, variables: Variables): RenderResult<T> => {
   try {
     const { wrappedTemplate, usedVariables } = collectAndEscapeVariablesFromTemplate(input, variables);
     const hbsTemplate = compile(wrappedTemplate);
-    const renderedTemplate = hbsTemplate(variables);
+    const renderedTemplate = hbsTemplate(variables) as T;
```

Comment:
`as` assertions should generally be avoided if possible. since it wasn't possible in this case I had added reasoning for it in the [reference code](https://github.com/requestly/requestly/pull/2969) could you please add those as well